### PR TITLE
Feat/auth integration

### PR DESCRIPTION
**Pull request by martinradadl**

### CODE REVIEW

**Comment by cristiancho29** [comment]
Im seeing that the transition go to a white background. Maybe you can set a blue navy background to the root

https://github.com/user-attachments/assets/0316339b-7987-482c-b129-db44482e2b29

**Comment by cristiancho29** [comment]
User should redirected to "/" after a succesful login o register, user shouldn't be recharging the page to know that is logged. It seems like nothing happen

https://github.com/user-attachments/assets/cadaa0b2-007d-49d7-8da9-b00c14db593d

File: src/pages/auth/login.tsx [hunk 1]
```diff
@@ -1,9 +1,19 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
+import { useAuth } from "../../data/authentication";
+import { useCookies } from "react-cookie";
 
 export const Login: React.FC = () => {
   const [user, setUser] = useState({ email: "", password: "" });
   const navigate = useNavigate();
+  const { login } = useAuth();
```
Comment (cristiancho29) [inline src/pages/auth/login.tsx:9]: To avoid repeating the redirection you could create empty wrapper to only manage the logic of redirection on login and register, we need to reduce redundant code

File: src/pages/auth/sign-up.tsx [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
+import { useAuth } from "../../data/authentication";
+import { useCookies } from "react-cookie";
 
 export const SignUp: React.FC = () => {
   const [user, setUser] = useState({ name: "", email: "", password: "" });
   const navigate = useNavigate();
+  const { register } = useAuth();
+  const [cookies] = useCookies(["user"]);
```
Comment (cristiancho29) [inline src/pages/auth/sign-up.tsx]: Same as login page

File: src/pages/dashboard.tsx [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+import React, { useEffect } from "react";
+import { useCookies } from "react-cookie";
+import { useNavigate } from "react-router-dom";
 
 export const Dashboard: React.FC = () => {
-    return <h1>Dashboard</h1>;
-}
\ No newline at end of file
+  const [cookies] = useCookies(["user"]);
+  const navigate = useNavigate();
```
Comment (cristiancho29) [inline src/pages/dashboard.tsx]: Move this logic to MainLayout component

File: src/pages/debts.tsx [hunk 4]
```diff
@@ -1,5 +1,16 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { useCookies } from "react-cookie";
+import { useNavigate } from "react-router-dom";
 
 export const Debts: React.FC = () => {
+  const [cookies] = useCookies(["user"]);
+  const navigate = useNavigate();
```
Comment (cristiancho29) [inline src/pages/debts.tsx]: Same as Dashboard comment

File: src/pages/profile.tsx [hunk 5]
```diff
@@ -1,6 +1,16 @@
...
-
+import React, { useEffect } from "react";
+import { useCookies } from "react-cookie";
+import { useNavigate } from "react-router-dom";
 
 export const Profile: React.FC = () => {
-    return <h1>Profile</h1>;
-}
\ No newline at end of file
+  const [cookies] = useCookies(["user"]);
```
Comment (cristiancho29) [inline src/pages/profile.tsx]: Same as Dashboard comment

File: src/pages/transactions/index.tsx [hunk 6]
```diff
@@ -12,6 +12,8 @@ import {
   TransactionFormI,
 } from "../../data/transactions";
 import { DeleteTransactionModal } from "./delete-modal";
+import { useNavigate } from "react-router-dom";
+import { useCookies } from "react-cookie";
```
Comment (cristiancho29) [inline src/pages/transactions/index.tsx]: Same as Dashboard comment

File: src/tests/authentication.test.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+const createWrapper = () => {
+  const wrapper = ({ children }: { children: React.ReactNode }) => (
+    <RecoilRoot>{children}</RecoilRoot>
+  );
+  return wrapper;
+};
+
+const wrapper = createWrapper();
+
+test("register", async () => {
```
Comment (cristiancho29) [inline src/tests/authentication.test.tsx]: Missing the test where status is not 200, you can expect that the status is different that 200 and that the user is null

File: src/tests/authentication.test.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+  const { result } = renderHook(() => useAuth(), { wrapper });
+  await act(async () => {
+    result.current.register(newUser);
+  });
+  expect(result.current.user).toEqual(newUser);
+});
+
+test("login", async () => {
+  vi.mocked(axios, true).post.mockResolvedValueOnce({
```
Comment (cristiancho29) [inline src/tests/authentication.test.tsx]: Same as the missing register test

File: src/tests/authentication.test.tsx [hunk 9]
```diff
@@ -0,0 +1,91 @@
+import { renderHook, act } from "@testing-library/react";
```
Comment (cristiancho29) [inline src/tests/authentication.test.tsx:1]: Let's try to add the describes at once, to avoid adding that change later

**Review by cristiancho29 — CHANGES_REQUESTED** [review]

File: src/data/authentication.tsx [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+export const userState = atom<UserI | null>({
+  key: "userState",
+  default: null,
+});
+
+export const useAuth = () => {
+  const [user, setUser] = useRecoilState(userState);
+  const [userCookie, setCookie /*removeCookie*/] = useCookies(["user"]);
```
Comment (cristiancho29) [inline src/data/authentication.tsx]: Remove the comented code

**Review by cristiancho29 — CHANGES_REQUESTED** [review]

**Review by cristiancho29 — APPROVED** [review]
LGTM
